### ISSUE TITLE
Lurker terminated with 'OOMKilled' event

### ISSUE THREAD

**Issue by Alon-Katz** [secureCodeBox/secureCodeBox#2327]
### Describe the bug
The lurker is failing due to an 'OOMKilled' issue and the resource allocations are hard-coded in the code, which raises the question: why? In heavy production environments, it is unreasonable to be constrained by the following resource limits without any option to increase them:
```
    Limits:
      cpu:     100m
      memory:  100Mi
    Requests:
      cpu:     20m
      memory:  20Mi
```

### Steps To Reproduce
Run a scan with thousands of findings

### Expected behavior
Please either implement a method to allow the increase of resources allocated to the lurker or ensure that it cannot be terminated due to an OOM event.

### System (please complete the following information):
secureCodeBox Version/Release:
v4.4.1
Kubernetes Version [command: kubectl version]:
Client Version: v1.29.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.5-eks-5e0fdde

### Screenshots / Logs
```
  containerStatuses:
  - containerID: containerd://f0455a853556f2174458a2320c4d4f5fc68d271b96788a78eff16c67d4dfed02
    image: docker.io/zricethezav/gitleaks:v8.18.2
    lastState: {}
    name: gitleaks
    ready: false
    restartCount: 0
    started: false
    state:
      terminated:
        containerID: containerd://f0455a853556f2174458a2320c4d4f5fc68d271b96788a78eff16c67d4dfed02
        exitCode: 0
        finishedAt: "2024-03-03T10:05:36Z"
        reason: Completed
        startedAt: "2024-03-03T10:04:33Z"
  - containerID: containerd://9ad202c0521385c90259273ef5576289f01ec3bb61d38c4e862a97ad59e98fdb
    image: docker.io/securecodebox/lurker:4.4.1
    lastState:
      terminated:
        containerID: containerd://9ad202c0521385c90259273ef5576289f01ec3bb61d38c4e862a97ad59e98fdb
        exitCode: 137
        finishedAt: "2024-03-03T10:06:04Z"
        reason: OOMKilled
        startedAt: "2024-03-03T10:06:00Z"
    name: lurker
    ready: false
    restartCount: 2
    started: false
    state:
      waiting:
        message: back-off 20s restarting failed container=lurker pod=scan-retry-2f1cf8e1-8bf5-4e7e-a0de-93fe6ac44005-2prlb-4sj5g_scb-scanners(05be3608-393d-437e-8386-89422b52d0c6)
        reason: CrashLoopBackOff  
```

**Comment by J12934** [comment]
HI @Alon-Katz 

Do you know how big the file is that it's trying to upload there?

The lurker should stream the upload and never load the complete file into memory, so I don't know where this would happen.
We been using the lurker with these resource limits for a ton of scans and never ran into this issues even when uploading some really big files.

**Comment by Alon-Katz** [comment]
Hi @J12934 , file size was 20.6 Mb, and it's easily reproduced each time.
I think the best option here is to allow customization on the lurker resources

**Comment by J12934** [comment]
Had a look at the code, had this falsely remember it is loading the entire file in memory.
Should hopefully be easy to change / fix it to avoid these issues alltogether.

I don't quit agree that making this configurable is the best fix for it. That would just be one additional thing people would need to tweak to ensure the systems works properly, I'd rather try to see if we can avoid the issue alltogether.

That being said, of course having the resources for the lurker be configurable should still be possible, just think we should ensure that the default is working properly, vene for large files.

**Comment by J12934** [comment]
Should be fixed with release v4.5.0 :)
Can you check if this resolves the issue for you?

**Comment by Alon-Katz** [comment]
Hi @J12934 , Yes seems like it resolved the issue, thank you !
